runtime/oti/VMHelpers.hpp

```diff
@@ -2151,6 +2151,29 @@ class VM_VMHelpers
 		}
 	}
 
+	static VMINLINE void
+	enterVThreadTransitionCritical(J9VMThread *currentThread, jobject thread)
+	{
+		J9JavaVM *vm = currentThread->javaVM;
+		J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+		MM_ObjectAccessBarrierAPI objectAccessBarrier = MM_ObjectAccessBarrierAPI(currentThread);
+		j9object_t threadObj = J9_JNI_UNWRAP_REFERENCE(thread);
+
+		while(!objectAccessBarrier.inlineMixedObjectCompareAndSwapU64(currentThread, threadObj, vm->virtualThreadInspectorCountOffset, 0, (U_64)-1)) {
```

Comment:
Note that asserts in header files are problematic since we don't know what module we're in.

Reply:
Perhaps these should not be inline functions at all.

Reply:
enter/exit is currently only used in `javanextvmi.cpp` atm, I will move them into `javanextvmi` as static helper calls

Reply:
also per previous investigation, there might be a possibility that -1 would be set when enter is attempted. reason for this is that as soon as a vthread have been yielded (state changed) even before the unmountEnd notification have been called, it may be rescheduled to run on another carrier thread which would try enter critical region. so assert here wouldn't be correct